modules/idLibrary.js

```diff
@@ -0,0 +1,238 @@
+const OBSERVER_CONFIG = {
+  subtree: true,
+  attributes: true,
+  attributeOldValue: false,
+  childList: true,
+  attirbuteFilter: ['value'],
+  characterData: true,
+  characterDataOldValue: false
+};
+const logInfo = createLogInfo(LOG_PRE_FIX);
+const logError = createLogError(LOG_PRE_FIX);
+
+function createLogInfo(prefix) {
+  return function (...strings) {
+    utils.logInfo(prefix + ' ', ...strings);
+  }
+}
+
+function createLogError(prefix) {
+  return function (...strings) {
+    utils.logError(prefix + ' ', ...strings);
+  }
+}
+
+function getEmail(value) {
+  const matched = value.match(/([a-zA-Z0-9._-]+@[a-zA-Z0-9._-]+\.[a-zA-Z0-9._-]+)/gi);
+  if (!matched) {
+    return null;
+  }
+  logInfo('Email found' + matched[0]);
+  return matched[0];
+}
+
+function bodyAction(conf, mutations, observer) {
+  logInfo('BODY observer on debounce called');
+
+  if (email) {
+    observer.disconnect();
+    logInfo('Email is found, body observer disconnected');
+  }
+
+  const body = document.body.innerHTML;
+  email = getEmail(body);
+  if (email != null) {
+    logInfo(`Email obtained from the body ${email}`);
+    observer.disconnect();
+    logInfo('Post data on email found in body');
+    postData(conf.url);
+  }
+}
+
+function targetAction(conf, mutations, observer) {
+  logInfo('Target observer called');
+  for (const mutation of mutations) {
+    for (const node of mutation.addedNodes) {
+      email = node.textContent;
+
+      if (email) {
+        logInfo('Email obtained from the target ' + email);
+        observer.disconnect();
+        logInfo(' Post data on email found in target');
+        postData(conf.url);
+      }
+    }
+  }
+}
+
+function addInputElementsElementListner(conf) {
+  logInfo('Adding input element listeners');
+  const inputs = document.querySelectorAll('input[type=text], input[type=email]');
+
+  for (var i = 0; i < inputs.length; i++) {
+    logInfo(` Original Value in Input = ${inputs[i].value}`);
+    inputs[i].addEventListener('change', event => processInputChange(event, conf));
+    inputs[i].addEventListener('blur', event => processInputChange(event, conf));
+  }
+}
+
+function removeInputElementsElementListner(conf) {
+  logInfo('Removing input element listeners');
+  const inputs = document.querySelectorAll('input[type=text], input[type=email]');
+
+  for (var i = 0; i < inputs.length; i++) {
+    inputs[i].removeEventListener('change', event => processInputChange(event, conf));
+    inputs[i].removeEventListener('blur', event => processInputChange(event, conf));
+  }
+}
+
+function processInputChange(event, conf) {
+  const value = event.target.value;
+  logInfo(`Modified Value of input ${event.target.value}`);
+  email = getEmail(value);
+  if (email != null) {
+    logInfo('Email found in input ' + email);
+    postData(conf.url);
+    removeInputElementsElementListner(conf);
+  }
+}
+
+function debounce(func, wait, immediate) {
+  let timeout;
+
+  return function (...args) {
```

Comment:
So debounce is just returning a function to be called later?

However, anywhere debounce is called, the return is not used.

So this is effectively not doing anything.

I think the real fix here is to add the method bindings as requested for the other methods (first param anytime debounce is called).

And to make this function a regular function that just executes, not one which returns a function.

Reply:
Also, did you mean to use the keyword `arguments` instead of `args` ?

Reply:
@robertrmartinez The debounce issue is fixed and full body scan config is provided.